hexbytes/main.py

```diff
@@ -29,5 +30,20 @@ def hex(self) -> str:
         '''
         return '0x' + super().hex()
 
+    @overload
+    def __getitem__(self, key: int) -> int:
+        ...
+
+    @overload  # noqa: F811
+    def __getitem__(self, key: slice) -> 'HexBytes':
+        ...
+
+    def __getitem__(self, key: Union[int, slice]) -> Union[int, bytes, 'HexBytes']:  # noqa: F811
+        result = super().__getitem__(key)
+        if hasattr(result, 'hex'):
```

Comment:
maybe `try/except` instead of checking?

Reply:
Let me run some performance tests... Not sure how bad it is to throw an exception every time that you try to get a single byte out.

Reply:
~3% faster for the slice case, and ~20% slower for the index to one byte case. Seems like it's better to leave as is.